src/dilep_producer.cpp

```diff
@@ -30,38 +35,39 @@ void DileptonProducer::WriteDileptons(pico_tree &pico,
                          pico.out_mu_phi()[imu2],0.10566);
         dimu = mu1 + mu2;
         // Dilepton closest to Z mass gets put at the front
-        if(abs(dimu.M() - zmass) < mindm) { 
+        if((abs(dimu.M() - zmass) < mindm) && (pico.out_mu_charge()[imu1]+pico.out_mu_charge()[imu2])==0) {
```

Comment:
I think using the nano.Muon_charge() would be better?

Reply:
pico.mu_charge is storing the same information. Is there any reason we would want to change it?
Also, I am not passing nano tree to dilep_producer, so only the pico variables are used in this producer.

Reply:
Okay. I guess it's a matter of style. I like having clean inputs and outputs. I was just worried that if the pico muon sequence is changed or if a nano muon is skipped, that it would cause a problem. 
But I guess it is fine.